src/Microsoft.AspNetCore.Server.Kestrel/KestrelServerOptions.cs

```diff
@@ -6,16 +6,35 @@
 
 namespace Microsoft.AspNetCore.Server.Kestrel
 {
+    /// <summary>
+    /// Provides programmatic configuration of Kestrel-specific features.
+    /// </summary>
     public class KestrelServerOptions
     {
+        /// <summary>
+        /// Typically initialized by <see cref="Hosting.WebHostBuilderKestrelExtensions.UseKestrel(Hosting.IWebHostBuilder, Action{KestrelServerOptions})"/>.
+        /// Enables the UseKestrel options callback to resolve and use services registered by the application during startup.
+        /// </summary>
         public IServiceProvider ApplicationServices { get; set; }
 
+        /// <summary>
+        /// Gets or sets an <see cref="IConnectionFilter"/> that allows each connection <see cref="System.IO.Stream"/>
+        /// to be intercepted and transformed.
+        /// Configured by the UseHttps() and <see cref="Hosting.KestrelServerOptionsConnectionLoggingExtensions.UseConnectionLogging(KestrelServerOptions)"/>
```

Comment:
It's in an unreferenced package :cry: I still think it's worth mentioning, but cref throws up if I try to use it.